File: csharp/resistor-color-trio/ResistorColorTrio.cs
using System;

public static class ResistorColorTrio
{
    private enum ColorCodes
    {
        black,
        brown,
        red,
        orange,
        yellow,
        green,
        blue,
        violet,
        grey,
        white,
    }
    public static int ColorCode(string color)
    {
        // can even make it case insensitive with true as second argument.
        return (int)Enum.Parse<ColorCodes>(color, true);
    }
    public static int Value(string[] colors)
    {
        int val = 10 * ColorCode(colors[0]) + ColorCode(colors[1]);
        for (int i = 0; i < ColorCode(colors[2]); ++i) val *= 10;
        return val;
    }
    public static string Label(string[] colors)
    {
        int val = Value(colors);
        bool divisibleByThousand = val % 1000 == 0;
        return divisibleByThousand ? (val / 1000).ToString() + " kiloohms" : val.ToString() + " ohms"; 
    }
}

File: fsharp/resistor-color-trio/ResistorColorTrio.fs
module ResistorColorTrio

let allColors: string list = ["black"; "brown"; "red"; "orange"; "yellow"; "green"; "blue"; "violet"; "grey"; "white"]

let exponents: int list = [1; 10; 100; 1000; 10000; 100000; 1000000; 10000000; 100000000; 1000000000]

let colorCode (color: string) : int = allColors |> List.findIndex ((=) color)

let value (colors : string list) =  ((colorCode colors[0]) * 10 + (colorCode colors[1])) * (exponents[colorCode colors[2]])

let label colors = 
    let value = value colors
    if value % 1000000 = 0 then sprintf "%d megaohms" (value / 1000000)
    elif value % 1000 = 0 then sprintf "%d kiloohms" (value / 1000)
    else sprintf "%d ohms" value
